WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2341,42 +2341,14 @@ private MediaModel queueFileForUpload(Uri uri, String mimeType, MediaUploadState
     }
 
     private MediaModel buildMediaModel(Uri uri, String mimeType, MediaUploadState startingState) {
-        String path = MediaUtils.getRealPathFromURI(this, uri);
-
-        MediaModel media = mMediaStore.instantiateMediaModel();
-        AppLog.i(T.MEDIA, "New media instantiated localId=" + media.getId());
-        String filename = org.wordpress.android.fluxc.utils.MediaUtils.getFileName(path);
-        String fileExtension = org.wordpress.android.fluxc.utils.MediaUtils.getExtension(path);
-
-        // Try to get mimetype if none was passed to this method
-        if (mimeType == null) {
-            mimeType = getContentResolver().getType(uri);
-            if (mimeType == null) {
-                mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(fileExtension);
-            }
-            if (mimeType == null) {
-                // Default to image jpeg
-                mimeType = "image/jpeg";
-            }
-        }
-        // If file extension is null, upload won't work on wordpress.com
-        if (fileExtension == null) {
-            fileExtension = MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType);
-            filename += "." + fileExtension;
-        }
 
-        if (org.wordpress.android.fluxc.utils.MediaUtils.isVideoMimeType(mimeType)) {
+        MediaModel media = FluxCUtils.mediaModelFromLocalUri(this, uri, mimeType, mMediaStore, mSite.getId());
+        if (org.wordpress.android.fluxc.utils.MediaUtils.isVideoMimeType(media.getMimeType())) {
```

Comment:
Do you think setting the video thumbnail should be done in `mediaModelFromLocalUri()` instead of here, so it's always set?

Reply:
I’ve been looking for usages of `MediaModel.getThumbnailUrl` and it seems, if we do that, we’d need to refactor  
- `loadVideoThumbnail` in `MediaGridAdapter`, 
- `downloadVideoThumbnail` in `MediaSettingsActivity`
- and probably look better into how it’s used in `updateFeaturedImageView` in `EditPostSettingsFragment`

I see the benefits of having clearer code (and less repetition) with that, but I think it might be good to do on another PR as the change might be a bit bigger than it initially seems.

Reply:
Agreed, let's do that separately.